systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -678,6 +680,59 @@ void testScaleMirrorMaker2ToZero() {
         assertNull(mm2Status.getUrl());
     }
 
+    @Test
+    void testIdentityReplicationPolicy() {
+        String originalTopicName = "original-topic";
+
+        // Deploy source kafka
+        KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 1, 1).done();
+        // Deploy target kafka
+        KafkaResource.kafkaEphemeral(kafkaClusterTargetName, 1, 1).done();
+        // Create topic
+        KafkaTopicResource.topic(kafkaClusterSourceName, originalTopicName, 3).done();
+
+        KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+        final String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+        KafkaMirrorMaker2Resource.kafkaMirrorMaker2(CLUSTER_NAME, kafkaClusterTargetName, kafkaClusterSourceName, 1, false)
+            .editSpec()
+                .editMirror(0)
+                    .editSourceConnector()
+                        .addToConfig("replication.policy.class", "io.strimzi.kafka.connect.mirror.IdentityReplicationPolicy")
+                    .endSourceConnector()
+                .endMirror()
+            .endSpec()
+            .done();
+
+        LOGGER.info("Sending and receiving messages via {}", kafkaClusterSourceName);
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withNamespaceName(NAMESPACE)
+            .withTopicName(originalTopicName)
+            .withClusterName(kafkaClusterSourceName)
+            .withMessageCount(MESSAGE_COUNT)
+            .withUsingPodName(kafkaClientsPodName)
+            .build();
+
+        internalKafkaClient.assertSentAndReceivedMessages(
+            internalKafkaClient.sendMessagesPlain(),
+            internalKafkaClient.receiveMessagesPlain()
+        );
+
+        LOGGER.info("Changing to {} and will try to receive messages", kafkaClusterTargetName);
+        internalKafkaClient.setClusterName(kafkaClusterTargetName);
+
+        assertThat(internalKafkaClient.receiveMessagesPlain(), equalTo(MESSAGE_COUNT));
+
+        LOGGER.info("Checking if the mirrored topic name is same as the original one");
+
+        KafkaTopic kafkaTopic = KafkaTopicResource.kafkaTopicClient().inNamespace(NAMESPACE).withName(originalTopicName).get();
+        assertNotNull(kafkaTopic);
+        assertThat(kafkaTopic.getMetadata().getName(), equalTo(originalTopicName));
+        assertThat(kafkaTopic.getSpec().getPartitions(), equalTo(3));
+        assertThat(kafkaTopic.getSpec().getReplicas(), equalTo(1));
```

Comment:
Stupid question ... but ... On line 692 you create topic `original-topic` and here you just list it and check it has the right name? Because how do you know this is not the initial topic you created when using single namespace? I guess you could say that if on line 725 you can read the mirrored messages from the topic `original-topic` it worked. But this should be either fixed or removed.

Reply:
Yes, that's true, I'm looking at solution that getting (somehow, if it is possible) the `KafkaTopics` from MM2 API ...

Reply:
But :smile: don't know if it is possible

Reply:
Nothing easy comes to my mind. I don't even know how the TO would handle it. So I guess you can:
* decide that reading the messages from the topic is good enough (assuming the internal client doesn't subscribe to some regex)
* Check it using Kafka APIs / kafka_topics.sh

Reply:
I was thinking about it, I will check it via Kafka APIs/kafka_topics.sh -> I want to be 100% sure of this :smile:

Reply:
Now it should be more sufficient